### PR TITLE
Add locale selector to email preview

### PR DESCRIPTION
**Pull request by plus3x**
@rafaelfranca Fix of #19922

### CODE REVIEW

File: railties/lib/rails/application_controller.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@ class Rails::ApplicationController < ActionController::Base # :nodoc:
   self.view_paths = File.expand_path('../templates', __FILE__)
   layout 'application'
 
+  before_action :set_locale
```
Comment (egilburg) [inline railties/lib/rails/application_controller.rb]: Is this going to generic ApplicationController? It's an unrelated change to emails, although perhaps useful to discuss in it's own right.

**Comment by plus3x** [comment]
@egilburg Rails::MailersController inherited from Rails::ApplicationController and use locale. I hope Rails::WelcomeController and Rails::InfoController can also be translated.

**Comment by rafaelfranca** [comment]
> I hope Rails::WelcomeController and Rails::InfoController can also be translated.

This will not happen. I can see reasons to accept different locales in mail previews since it change the result of the email body but not in Welcome and Info controllers.

Let move this behavior to `Rails::MailersController`

**Comment by repinel** [comment]
Please squash the commits. Thanks!

**Comment by kamipo** [comment]
Closing in favor of #31596.
